launchpod/angular-launchpod/backend/src/test/java/com/google/launchpod/FileUploadServletTest.java

```diff
@@ -0,0 +1,666 @@
+package com.google.launchpod;
+
+import static com.google.appengine.api.datastore.FetchOptions.Builder.withLimit;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.beans.Transient;
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.security.interfaces.DSAKey;
+import java.text.SimpleDateFormat;
+import java.text.ParseException;
+import java.util.Date;
+
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.launchpod.servlets.FileUploadServlet;
+import com.google.launchpod.data.UserFeed;
+import com.google.launchpod.data.RSS;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EmbeddedEntity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.cloud.storage.testing.RemoteStorageHelper;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.After;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.joda.time.DateTimeUtils;
+
+/**
+ * Runs unit tests for the FileUploadServlet that contains doPost() and doGet() methods.
+ */
+@RunWith(JUnit4.class)
+public class FileUploadServletTest extends Mockito {
+
+  @InjectMocks
+  private FileUploadServlet servlet = new FileUploadServlet();
+
+  @Mock
+  HttpServletRequest request;
+
+  @Mock
+  HttpServletResponse response;
+
+  @Rule // JUnit 4 uses Rules for testing specific messages
+  public ExpectedException thrown = ExpectedException.none();
+
+  private final LocalServiceTestHelper helper = new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  // Keys
+  private static final String USER_FEED = "UserFeed";
```

Comment:
are these constants also in the main code? if so you should reference them instead of copying

Reply:
will be done in a separate PR specifically meant for common methods and constants